shotgun_api3/shotgun.py

```diff
@@ -2206,6 +2333,58 @@ def upload(self, entity_type, entity_id, path, field_name=None, display_name=Non
         attachment_id = int(str(result).split(":")[1].split("\n")[0])
         return attachment_id
 
+    def _get_attachment_upload_info(self, is_thumbnail, filename):
+        """
+        Internal function to get the information needed to upload a file to Cloud storage.
+
+        :param bool is_thumbnail: indicates if the attachment is a thumbnail.
+        :param str filename: name of the file that will be uploaded.
+
+        :returns: dictionary containing the ulpload_info and the url to upload to.
```

Comment:
Is upload info the data that you are passing back to the server in "pass 3" of the upload process? Might be worth adding a sentence to explain what it is here.